docs/index.md

```diff
@@ -306,3 +306,7 @@ To get started, see the [Provider Example Configs](https://github.com/couchbasec
 
   Manage Flush Bucket
 
+* [Manage GSI](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/gsi)
+
+  Manage GSI
```

Comment:
@l0n3star Tagging in reference to discussion on slack thread. Tagging here so we don't miss

Reply:
@a-atri @simon-dew 

please note I made significant updates to the non-deferred and deferred examples for several reasons.   first, had to include output showing index status.  this is priority for priya.  second, updated the examples to give better guidance to users.  in particular, we should encourages users to create 1 non-deferred index.  third, priya wanted to use different bucket/scope/collection names to make it more user friendly.

please reference updated examples below:
https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/7147d366f6fffee59b4772e16bf910e33dc9e680/examples/gsi